routes/auth.routes.js

```diff
@@ -0,0 +1,102 @@
+const saltRounds = 10;
+
+// require auth middleware
+const { isLoggedIn, isLoggedOut } = require('../middleware/route-guard.js');
+
+// GET route ==> to display the signup form to users
+router.get('/signup', isLoggedOut, (req, res) => res.render('auth/signup'));
+
+// POST route ==> to process form data
+router.post('/signup', isLoggedOut, (req, res, next) => {
+    //console.log('The form data: ', req.body);
+
+    const { username, password } = req.body;
+
+    if(!username || !password) {
+      res.render('auth/signup', { errorMessage: 'All fields are mandatory. Please provide your username and password.' });
+      return;
+    }
+
+    const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+    if (!regex.test(password)) {
+      res
+        .status(500)
+        .render('auth/signup', { errorMessage: 'Password needs to have at least 6 chars and must contain at least one number, one lowercase and one uppercase letter.' });
+      return;
+    }
+
+    bcryptjs
+    .genSalt(saltRounds)
+    .then(salt => bcryptjs.hash(password, salt))
+    .then(hashedPassword => {
+        return User.create({
+            username,
+            password: hashedPassword
+          });
+    })
+    .then(() => {
+        //console.log('Newly created user is: ', userFromDB);
+        res.redirect('/userProfile');
+      })
+    .catch(error => {
+      if (error instanceof mongoose.Error.ValidationError) {
+        res.status(500).render('auth/signup', {errorMessage: error.message})
+      } else if (error.code = 11000) {
+        res.render('auth/signup', {errorMessage: 'Username must be unique. This username is already used.'})
+      } else {
+        next(error)
+      }
+      });
+  });
+
+// GET login route ==> to display the login form to users
+router.get('/login', isLoggedOut, (req, res) => res.render('auth/login'));
+
+// POST login route ==> to process form data
+router.post('/login', isLoggedOut, (req, res, next) => {
+  console.log('SESSION =====> ', req.session);
+  const { username, password } = req.body;
+ 
+  if (username === '' || password === '') {
+    res.render('auth/login', {
+      errorMessage: 'Please enter both, username and password to login.'
+    });
+    return;
+  }
+ 
+  User.findOne({ username })
+    .then(user => {
+      if (!user) {
+        res.render('auth/login', { errorMessage: 'Username is not registered. Try with other username.' });
+        return;
+      } else if (bcryptjs.compareSync(password, user.password)) {
+        req.session.currentUser = user;
+        res.redirect('/userProfile');
+      } else {
+        res.render('auth/login', { errorMessage: 'Incorrect password.' });
+      }
+    })
+    .catch(error => next(error));
+});
+
+router.get('/userProfile', isLoggedIn, (req, res) => res.render('users/user-profile', { userInSession: req.session.currentUser }));
+
+// /main route
+
+router.get('/main', isLoggedIn, (req, res) => res.render('/main', { userInSession: req.session.currentUser }));
```

Comment:
you the view you are trying to render is not inside a folder, do not use a leading `/` (the `views` folder does not count)